argonaut/src/test/scala/pathy/argonaut/PosixCodecJsonSpec.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014 - 2015 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pathy.argonaut
+
+import pathy.Path, Path._
+import pathy.scalacheck.PathyArbitrary._
+
+import argonaut._
+import org.specs2.mutable
+import org.specs2.ScalaCheck
+
+final class PosixCodecJsonSpec extends mutable.Specification with ScalaCheck {
+  import PosixCodecJson._
+
+  "RelFile Codec" ! prop { rf: RelFile[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[RelFile[Unsandboxed]])(unsandbox(rf))
+  }
+
+  "RelDir Codec" ! prop { rd: RelDir[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[RelDir[Unsandboxed]])(unsandbox(rd))
+  }
+
+  "AbsFile Codec" ! prop { af: AbsFile[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[AbsFile[Sandboxed]])(af)
+  }
+
+  "AbsDir Codec" ! prop { ad: AbsDir[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[AbsDir[Sandboxed]])(ad)
+  }
+}
```

Comment:
I'd like to see tests of error behavior, e.g. `(f: RelFile[Sandboxed]) => (EncodeJson.of[RelFile].encode(rf).as[AbsFile]`